### PR TITLE
Micro-XRCE-DDS client integration

### PR DESCRIPTION
**Pull request by jnippula**
Replace micrortps client with micro-xrce-dds client.

micrortps bridge is deprecated and support will be end in some point. We need to switch to use new micro-XRCE-DDS bridge for topic sharing between PX4 and ROS2

### CODE REVIEW

**Review by jlaitine — APPROVED** [review]
Good stuff!

File: boards/px4/fmu-v5x/ssrc.px4board [hunk 1]
```diff
@@ -11,6 +11,20 @@ CONFIG_SYSTEMCMDS_REFLECT=n
...
 CONFIG_SYSTEMCMDS_SD_STRESS=n
 CONFIG_SYSTEMCMDS_SERIAL_TEST=n
+CONFIG_DRIVERS_CAMERA_CAPTURE=n
+CONFIG_DRIVERS_CAMERA_TRIGGER=n
+CONFIG_DRIVERS_CAMERA_FEEDBACK=n
+CONFIG_MODULES_ROVER_POS_CONTROL=n
+CONFIG_MODULES_SIH=n
+CONFIG_SYSTEMCMDS_NETMAN=n
+CONFIG_SYSTEMCMDS_REFLECT=n
+CONFIG_COMMON_DIFFERENTIAL_PRESSURE=n
```
Comment (jlaitine) [inline boards/px4/fmu-v5x/ssrc.px4board:21]: I think we will need some differential pressure sensor with fixed wings, so if possible leave this one

File: boards/px4/fmu-v5x/ssrc.px4board [hunk 2]
```diff
@@ -11,6 +11,20 @@ CONFIG_SYSTEMCMDS_REFLECT=n
...
+CONFIG_SYSTEMCMDS_SD_BENCH=n
+CONFIG_SYSTEMCMDS_SD_STRESS=n
+CONFIG_SYSTEMCMDS_SERIAL_TEST=n
+CONFIG_DRIVERS_ROBOCLAW=n
+CONFIG_MODULES_VTOL_ATT_CONTROL=n
```
Comment (jlaitine) [inline boards/px4/fmu-v5x/ssrc.px4board:26]: We may need VTOL control as well, so maybe leave this too

**Comment by jlaitine** [comment]
I added a few comments still; we have a requirement to fly fixed wings with Saluki; if possible lets keep this functionality also in pixhawk5x?

**Comment by jnippula** [comment]
According to comments, need to re-check the feature removals and then the flash image size after changes.

**Comment by jnippula** [comment]
Returned VTOL control and differential pressure sensor back. But due to flash size exceeding the memory area needed to make more cleanup:
Removed OSD driver and OPTICAL_FLOW features from Pixhawk5x config. @jlaitine is this Ok?

**Comment by jnippula** [comment]
Setting this PR on hold. Need to split changes to make image size reduction and microdds_client build first. Then the actual switch from micrortps to microXRCE-DDS can be done by just changing the call from startup script.
New PR here: https://github.com/tiiuae/px4-firmware/pull/182

**Review by jlaitine — APPROVED** [review]

**Pull request merged**